### PR TITLE
Hover card accessibility behavior

### PR DESCRIPTION
**Pull request by chaofeng2014**
#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [ ] Include a change request file using `$ npm run change`

#### Description of changes

1. Enable focus trap in hover card
2. set focus to hover card if focus trap is set
3. enable "ESC" key to close hover card.

#### Focus areas to test

(optional)

### CODE REVIEW

**Comment by erichdev** [comment]
@chaofeng2014 I made a commit directly to your branch and changed this line in `ExpandingCard.tsx` to fix the import:

from
`import { FocusTrapZone, IFocusTrapZone } from 'office-ui-fabric-react/lib/FocusTrapZone';`
to
`import { FocusTrapZone, IFocusTrapZone } from '../FocusTrapZone';`

That should fix the build error

File: packages/office-ui-fabric-react/src/components/HoverCard/ExpandingCard.tsx [hunk 1]
```diff
@@ -49,6 +53,10 @@ export class ExpandingCard extends BaseComponent<IExpandingCardProps, IExpanding
         needsScroll: true
       });
     }
+    this._events.on(this.refs.expandingCard, 'keydown', this._onDismiss);
```
Comment (atneik) [inline packages/office-ui-fabric-react/src/components/HoverCard/ExpandingCard.tsx]: Are we dismissing on every keydown?

File: packages/office-ui-fabric-react/src/components/HoverCard/HoverCard.tsx [hunk 2]
```diff
@@ -156,10 +163,13 @@ export class HoverCard extends BaseComponent<IHoverCardProps, IHoverCardState> {
   // Hide HoverCard
   @autobind
   private _cardDismiss(ev: MouseEvent) {
+    if (ev.type === 'keydown' && !(ev.which === KeyCodes.escape)) {
```
Comment (atneik) [inline packages/office-ui-fabric-react/src/components/HoverCard/HoverCard.tsx]: Why not have this check in `ExpandingCard.tsx`?

File: packages/office-ui-fabric-react/src/components/HoverCard/ExpandingCard.tsx [hunk 3]
```diff
@@ -65,6 +73,21 @@ export class ExpandingCard extends BaseComponent<IExpandingCardProps, IExpanding
     } = this.props;
     this._styles = getStyles(theme!, customStyles);
 
+    const content = (
+      <div
+        data-is-focusable={ true }
+        tabIndex={ 0 }
+        ref='expandingCard'
```
Comment (atneik) [inline packages/office-ui-fabric-react/src/components/HoverCard/ExpandingCard.tsx]: Why not use `ref={ this._resolveRef('_expandingCard')`?

File: packages/office-ui-fabric-react/src/components/HoverCard/ExpandingCard.tsx (hunk 1 above)
Comment (chaofeng2014) [reply packages/office-ui-fabric-react/src/components/HoverCard/ExpandingCard.tsx]: The check is in _onDismiss() function

Comment (chaofeng2014) [reply packages/office-ui-fabric-react/src/components/HoverCard/ExpandingCard.tsx]: Moved to ExpandingCard

File: packages/office-ui-fabric-react/src/components/HoverCard/HoverCard.tsx [hunk 4]
```diff
@@ -156,10 +163,13 @@ export class HoverCard extends BaseComponent<IHoverCardProps, IHoverCardState> {
   // Hide HoverCard
   @autobind
   private _cardDismiss(ev: MouseEvent) {
+    if (ev.type === 'keydown') {
```
Comment (atneik) [inline packages/office-ui-fabric-react/src/components/HoverCard/HoverCard.tsx]: is this still needed?

**Pull request merged**